Index: Makefile
===================================================================
RCS file: /cvsroot/gnokii/gnokii/Makefile,v
retrieving revision 1.118
diff -u -r1.118 Makefile
--- Makefile	23 Jul 2002 17:32:23 -0000	1.118
+++ Makefile	28 Aug 2002 15:45:23 -0000
@@ -25,19 +25,15 @@
 	common/links \
 	common/devices \
 	common/data \
+	po \
 	common \
 	$(BIN_DIRS)
 
 GTK_DIRS =  xgnokii
 
-PO_DIR   = 	po
 DOCS_DIR = 	Docs
 
 all: $(DIRS)
-	@if [ "x$(USE_NLS)" = xyes ]; then \
-		$(MAKE) -C $(PO_DIR); \
-	fi
-
 	@if [ "$(GTK_LIBS)" ]; then \
 		for dir in $(GTK_DIRS); do \
 		    if [ -e $$dir/Makefile ]; then \
@@ -66,9 +62,6 @@
 		$(MAKE) -C $$dir clean; \
 	    fi; \
 	done
-	@if [ "x$(USE_NLS)" = xyes ]; then \
-		$(MAKE) -C $(PO_DIR) clean; \
-	fi
 
 ifdef OWN_GETOPT
 		$(MAKE) -C getopt clean
@@ -88,9 +81,6 @@
 
 distclean: clean
 	$(MAKE) -C common distclean
-	@if [ -e $(PO_DIR)/Makefile ]; then \
-		$(MAKE) -C $(PO_DIR) distclean; \
-	fi
 	$(RM) Makefile.global config.cache config.log config.status \
 		include/config.h \
 		packaging/RedHat/gnokii.spec \
@@ -125,9 +115,6 @@
 			$(MAKE) -C $$dir install; \
 		fi; \
 	done
-	@if [ "x$(USE_NLS)" = xyes ]; then \
-		$(MAKE) -C $(PO_DIR) install; \
-	fi
 
 	@if [ "$(GTK_LIBS)" ]; then \
 		for dir in $(GTK_DIRS); do \
@@ -148,9 +135,6 @@
 			$(MAKE) -C $$dir install-strip; \
 		fi; \
 	done
-	@if [ "x$(USE_NLS)" = xyes ]; then \
-		$(MAKE) -C $(PO_DIR) install; \
-	fi
 
 	@if [ "$(GTK_LIBS)" ]; then \
 		for dir in $(GTK_DIRS); do \
@@ -167,9 +151,6 @@
 			$(MAKE) -C $$dir install-suid; \
 		fi; \
 	done
-	@if [ "x$(USE_NLS)" = xyes ]; then \
-		$(MAKE) -C $(PO_DIR) install; \
-	fi
 
 	@if [ "$(GTK_LIBS)" ]; then \
 		for dir in $(GTK_DIRS); do \
@@ -186,9 +167,6 @@
 			$(MAKE) -C $$dir install-ss; \
 		fi; \
 	done
-	@if [ "x$(USE_NLS)" = xyes ]; then \
-		$(MAKE) -C $(PO_DIR) install; \
-	fi
 
 	@if [ "$(GTK_LIBS)" ]; then \
 		for dir in $(GTK_DIRS); do \
Index: Makefile.global.in
===================================================================
RCS file: /cvsroot/gnokii/gnokii/Makefile.global.in,v
retrieving revision 1.31
diff -u -r1.31 Makefile.global.in
--- Makefile.global.in	27 Aug 2002 23:44:02 -0000	1.31
+++ Makefile.global.in	28 Aug 2002 15:45:23 -0000
@@ -56,7 +56,6 @@
 OWN_GETOPT     = @OWN_GETOPT@
 NO_SHARED      = @NO_SHARED@
 
-USE_NLS        = @USE_NLS@
 HAVE_XGNOKII   = @HAVE_XGNOKII@
 WIN32          = @WIN32@

Index: acconfig.h
===================================================================
RCS file: /cvsroot/gnokii/gnokii/acconfig.h,v
retrieving revision 1.4
diff -u -r1.4 acconfig.h
--- acconfig.h	27 Aug 2002 21:39:10 -0000	1.4
+++ acconfig.h	28 Aug 2002 15:45:23 -0000
@@ -51,9 +51,6 @@
 /* Define if you want security options enabled */
 #undef SECURITY
 
-/* Define if you want NLS support */
-#undef USE_NLS
-
 /* Use UNIX98 style pty support instead of the traditional */
 #undef USE_UNIX98PTYS
 
Index: autogen.sh
===================================================================
RCS file: /cvsroot/gnokii/gnokii/autogen.sh,v
retrieving revision 1.1
diff -u -r1.1 autogen.sh
--- autogen.sh	23 Apr 2002 23:19:26 -0000	1.1
+++ autogen.sh	28 Aug 2002 15:45:23 -0000
@@ -1,5 +1,6 @@
 #!/bin/sh
 
+aclocal
 autoheader
 autoconf
 ./configure "$@"
Index: configure.in
===================================================================
RCS file: /cvsroot/gnokii/gnokii/configure.in,v
retrieving revision 1.83
diff -u -r1.83 configure.in
--- configure.in	27 Aug 2002 23:44:02 -0000	1.83
+++ configure.in	28 Aug 2002 15:45:25 -0000
@@ -17,6 +17,7 @@
 
 dnl ======================== Default setting
 ALL_LINGUAS="cs de et fi nl sk pl it sl"
+AM_GNU_GETTEXT
 
 dnl ======================== Check for include/config.h.in
 AC_MSG_CHECKING([for include/config.h.in])
@@ -72,141 +73,6 @@
 	CFLAGS="-O2 -Wall"
 fi
 
-dnl ======================== NLS support
-
-LIBS=""
-NLS_LIBS=""
-NLS_CFLAGS=""
-
-AC_ARG_WITH(libintl,
-	[  --with-libintl=DIR      specifies the base NLS],
-	[ if test "x$withval" = "xyes"; then
-		AC_MSG_WARN(Usage is --with-libintl=DIR)
-	  else
-		NLS_LIBS="-L$withval/lib"
-		NLS_CFLAGS="-I$withval/include"
-	  fi
-	]
-)
-
-AC_ARG_ENABLE(nls,
-	[  --disable-nls           do not use NLS],
-	[ USE_NLS=$enableval ],
-	[ USE_NLS=yes ]
-)
-
-if test "$USE_NLS" = "yes"; then
-	AC_PROG_MAKE_SET
-	AC_CHECK_HEADERS(locale.h)
-	AC_CHECK_FUNCS(setlocale)
-	AC_SUBST(HAVE_LOCALE_H)  
-
-	CPPFLAGS="$CPPFLAGS $NLS_CFLAGS"
-	AC_CHECK_HEADERS(libintl.h,, USE_NLS=no)
-fi
-
-if test "$USE_NLS" = "yes"; then
-	AC_CHECK_FUNC(gettext,
-		      AC_DEFINE(HAVE_LIBINTL),
-		      AC_CHECK_LIB(intl, gettext,
-				   [ NLS_LIBS="$NLS_LIBS -lintl"
-				     AC_DEFINE(HAVE_LIBINTL) ],
-				   [ AC_MSG_WARN(not found library: intl !!!)
-				     USE_NLS=no ],
-				   $NLS_LIBS
-		      )
-	)
-fi
-
-if test "$USE_NLS" = "yes"; then
-
-	AC_SUBST(HAVE_LIBINTL_H)
-	AC_SUBST(HAVE_LIBINTL)
-
-	dnl Handle localedir
-	LOCALEDIR='${prefix}/share/locale'
-	AC_ARG_WITH(locale-dir,
-		    [  --with-locale-dir=DIR   Location of the locale file(s)
-			[PREFIX/share/locale]],
-		    [ if test x$withval = xyes; then
-			AC_MSG_WARN(Usage is: --with-locale-dir=basedir)
-		      else
-			if test x$withval = xno; then
-				AC_MSG_WARN(Usage is: --with-locale-dir=basedir)
-			else
-				LOCALEDIR=$withval
-			fi
-		      fi
-		    ]
-	)
-	AC_SUBST(LOCALEDIR)
-
-	AC_PATH_PROG(MSGFMT, msgfmt, $MSGFMT)
-	if test -n "$MSGFMT"; then
-		AC_CHECK_FUNCS(dcgettext)
-		AC_PATH_PROG(GMSGFMT, gmsgfmt, $MSGFMT)
-		AC_PATH_PROG(XGETTEXT, xgettext, $XGETTEXT)
-		AC_PATH_PROG(MSGMERGE, msgmerge, $MSGMERGE)
-
-		dnl Test whether we really found GNU xgettext.
-		if test -n "$XGETTEXT"; then
-			if $XGETTEXT --omit-header /dev/null 2> /dev/null; then
-				: ;
-			else
-				AC_MSG_RESULT([found xgettext program is not GNU xgettext; ignore it])
-				XGETTEXT=""
-			fi
-		fi
-
-		dnl We add another test for comparing GNU xgettext with openwin xgettext
-		if test -n "$XGETTEXT"; then
-			if $XGETTEXT --help > /dev/null 2> /dev/null; then
-				: ;
-			else
-				AC_MSG_RESULT([found xgettext program is not GNU xgettext; ignore it])
-				XGETTEXT=""
-			fi
-		fi
-
-		dnl Test whether we really found GNU msgfmt.
-		if test -n "$MSGFMT"; then
-			if $MSGFMT < /dev/null 2> /dev/null; then
-				AC_MSG_RESULT([found msgfmt program is not GNU msgfmt; NLS will not be installed])
-				MSGFMT=""
-			fi
-		fi
-
-		AC_TRY_LINK(,
-			    [ extern int _nl_msg_cat_cntr;
-			      return _nl_msg_cat_cntr],
-			    [ CATOBJEXT=.gmo
-			      DATADIRNAME=share],  
-			    [ CATOBJEXT=.mo
-			      DATADIRNAME=lib]
-		)
-		INSTOBJEXT=.mo
-	fi
-
-	if test -n "$ALL_LINGUAS"; then
-		for lang in $ALL_LINGUAS; do
-			CATALOGS="$CATALOGS $lang$CATOBJEXT"
-		done
-	fi
-	AC_SUBST(CATALOGS)
-	AC_SUBST(CATOBJEXT)
-	AC_SUBST(INSTOBJEXT)
-	AC_SUBST(DATADIRNAME)
-fi
-
-dnl don't use NLS, when there is no gettext installed
-if test x"$MSGFMT" = x; then
-	USE_NLS=no
-fi
-
-if test x"$USE_NLS" = xyes; then
-	AC_DEFINE(USE_NLS)
-fi
-
 dnl ======================== Checks for libraries.
 
 AC_ARG_WITH(libpthread, 
@@ -599,7 +465,6 @@
 AC_SUBST(XPACKAGE)
 AC_SUBST(XGNOKIIDIR)
 AC_SUBST(exec_prefix)
-AC_SUBST(USE_NLS)
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)

@@ -618,8 +483,7 @@
    Makefile.global
    po/Makefile.in
    packaging/RedHat/gnokii.spec
-   packaging/Slackware/SlackBuild,
-   [sed -e "/POTFILES =/r po/POTFILES" po/Makefile.in > po/Makefile]
+   packaging/Slackware/SlackBuild
    )
 
 
Index: gnokii/gnokii.c
===================================================================
RCS file: /cvsroot/gnokii/gnokii/gnokii/gnokii.c,v
retrieving revision 1.299
diff -u -r1.299 gnokii.c
--- gnokii/gnokii.c	27 Aug 2002 23:23:00 -0000	1.299
+++ gnokii/gnokii.c	28 Aug 2002 15:45:45 -0000
@@ -66,7 +66,7 @@
 #endif
 
 
-#ifdef USE_NLS
+#ifdef ENABLE_NLS
 #  include <locale.h>
 #endif
 
@@ -4534,7 +4534,7 @@
 	opterr = 0;
 
 	/* For GNU gettext */
-#ifdef USE_NLS
+#ifdef ENABLE_NLS
 	textdomain("gnokii");
 	setlocale(LC_ALL, "");
 #endif
Index: gnokiid/gnokiid.c
===================================================================
RCS file: /cvsroot/gnokii/gnokii/gnokiid/gnokiid.c,v
retrieving revision 1.27
diff -u -r1.27 gnokiid.c
--- gnokiid/gnokiid.c	18 Aug 2002 17:51:29 -0000	1.27
+++ gnokiid/gnokiid.c	28 Aug 2002 15:45:45 -0000
@@ -101,7 +101,7 @@
 	static bool atexit_registered = false;
 
 	/* For GNU gettext */
-#ifdef USE_NLS
+#ifdef ENABLE_NLS
 	textdomain("gnokii");
 #endif
 
Index: include/misc.h
===================================================================
RCS file: /cvsroot/gnokii/gnokii/include/misc.h,v
retrieving revision 1.45
diff -u -r1.45 misc.h
--- include/misc.h	17 Jul 2002 23:06:39 -0000	1.45
+++ include/misc.h	28 Aug 2002 15:45:46 -0000
@@ -109,14 +109,14 @@
 #endif
 
 /* This one is for NLS. */
-#ifdef USE_NLS
+#ifdef ENABLE_NLS
 #  include <libintl.h>
 #  define _(x) gettext(x)
 #  define N_(x) gettext_noop(x)
 #else
 #  define _(x) (x)
 #  define N_(x) (x)
-#endif /* USE_NLS */
+#endif /* ENABLE_NLS */
 
 /* Definitions for u8, u16, u32 and u64, borrowed from
    /usr/src/linux/include/asm-i38/types.h */
Index: po/POTFILES.in
===================================================================
RCS file: /cvsroot/gnokii/gnokii/po/POTFILES.in,v
retrieving revision 1.7
diff -u -r1.7 POTFILES.in
--- po/POTFILES.in	9 Nov 2001 13:55:53 -0000	1.7
+++ po/POTFILES.in	28 Aug 2002 15:45:47 -0000
@@ -14,7 +14,6 @@
 common/phones/nk6100.c
 common/phones/nk3110.c
 common/phones/nk2110.c
-common/vcal.c
 common/vcal.lx
 gnokiid/gnokiid.c
 gnokii/gnokii.c
Index: smsd/smsd.c
===================================================================
RCS file: /cvsroot/gnokii/gnokii/smsd/smsd.c,v
retrieving revision 1.25
diff -u -r1.25 smsd.c
--- smsd/smsd.c	18 Aug 2002 18:03:11 -0000	1.25
+++ smsd/smsd.c	28 Aug 2002 15:45:48 -0000
@@ -425,7 +425,7 @@
 
 int main (int argc, char *argv[])
 {
-#ifdef USE_NLS
+#ifdef ENABLE_NLS
   textdomain("gnokii");
 #endif
 
Index: utils/mgnokiidev.c
===================================================================
RCS file: /cvsroot/gnokii/gnokii/utils/mgnokiidev.c,v
retrieving revision 1.16
diff -u -r1.16 mgnokiidev.c
--- utils/mgnokiidev.c	23 Apr 2002 23:19:27 -0000	1.16
+++ utils/mgnokiidev.c	28 Aug 2002 15:45:49 -0000
@@ -148,7 +148,7 @@
 	char *name = NULL; /* Device name */
 
 	/* For GNU gettext */
-#ifdef USE_NLS
+#ifdef ENABLE_NLS
 	textdomain("gnokii");
 #endif
 
Index: win32/gnokii_vc6/config1
===================================================================
RCS file: /cvsroot/gnokii/gnokii/win32/gnokii_vc6/config1,v
retrieving revision 1.2
diff -u -r1.2 config1
--- win32/gnokii_vc6/config1	30 Apr 2002 18:56:12 -0000	1.2
+++ win32/gnokii_vc6/config1	28 Aug 2002 15:45:49 -0000
@@ -75,8 +75,8 @@
 #undef XPM
 
 /* Define if you want NLS */
-//#define USE_NLS 1
-#undef USE_NLS
+//#define ENABLE_NLS 1
+#undef ENABLE_NLS
 
 /* Define if you want to use Unix98 PTYs */
 //#define USE_UNIX98PTYS 1
Index: xgnokii/xgnokii.c
===================================================================
RCS file: /cvsroot/gnokii/gnokii/xgnokii/xgnokii.c,v
retrieving revision 1.56
diff -u -r1.56 xgnokii.c
--- xgnokii/xgnokii.c	27 Aug 2002 00:44:00 -0000	1.56
+++ xgnokii/xgnokii.c	28 Aug 2002 15:46:00 -0000
@@ -2322,7 +2322,7 @@
 
 int main(int argc, char *argv[])
 {
-#ifdef USE_NLS
+#ifdef ENABLE_NLS
 	textdomain("gnokii");
 #endif
 
