--- ./Docs/sample/gnokiirc.orig	Sun Jul  1 14:03:23 2001
+++ ./Docs/sample/gnokiirc	Sun Jul  1 14:03:44 2001
@@ -28,4 +28,4 @@
 # In particular ensure that mgnokiidev is in this location, with
 # permissions 4750, owned by root, group gnokii.  Ensure you
 # are in the gnokii group and that the group exists...
-bindir = /usr/local/sbin/
+bindir = /usr/sbin/
--- ./po/Makefile.in.in.orig	Sun Jul  1 14:10:49 2001
+++ ./po/Makefile.in.in	Sun Jul  1 14:13:27 2001
@@ -107,7 +107,7 @@
 install-data: install-data-@USE_NLS@
 install-data-no: all
 install-data-yes: all
-	${SHELL} $(top_srcdir)/mkinstalldirs $(datadir); \
+	${SHELL} $(top_srcdir)/mkinstalldirs $(DESTDIR)$(datadir); \
 	catalogs='$(CATALOGS)'; \
 	for cat in $$catalogs; do \
 	  case "$$cat" in \
@@ -116,22 +116,22 @@
 	  esac; \
 	  lang=`echo $$cat | sed 's/$(CATOBJEXT)$$//'`; \
 	  dir=$$destdir/$$lang/LC_MESSAGES; \
-	  ${SHELL} $(top_srcdir)/mkinstalldirs $$dir; \
+	  ${SHELL} $(top_srcdir)/mkinstalldirs $(DESTDIR)$$dir; \
 	  if test -r $$cat; then \
-	    $(INSTALL_DATA) $$cat $$dir/$(PACKAGE)$(INSTOBJEXT); \
+	    $(INSTALL_DATA) $$cat $(DESTDIR)$$dir/$(PACKAGE)$(INSTOBJEXT); \
 	    echo "installing $$cat as $$dir/$(PACKAGE)$(INSTOBJEXT)"; \
 	  else \
-	    $(INSTALL_DATA) $(srcdir)/$$cat $$dir/$(PACKAGE)$(INSTOBJEXT); \
+	    $(INSTALL_DATA) $(srcdir)/$$cat $(DESTDIR)$$dir/$(PACKAGE)$(INSTOBJEXT); \
 	    echo "installing $(srcdir)/$$cat as" \
 		 "$$dir/$(PACKAGE)$(INSTOBJEXT)"; \
 	  fi; \
 	  if test -r $$cat.m; then \
-	    $(INSTALL_DATA) $$cat.m $$dir/$(PACKAGE)$(INSTOBJEXT).m; \
+	    $(INSTALL_DATA) $$cat.m $(DESTDIR)$$dir/$(PACKAGE)$(INSTOBJEXT).m; \
 	    echo "installing $$cat.m as $$dir/$(PACKAGE)$(INSTOBJEXT).m"; \
 	  else \
 	    if test -r $(srcdir)/$$cat.m ; then \
 	      $(INSTALL_DATA) $(srcdir)/$$cat.m \
-		$$dir/$(PACKAGE)$(INSTOBJEXT).m; \
+		$(DESTDIR)$$dir/$(PACKAGE)$(INSTOBJEXT).m; \
 	      echo "installing $(srcdir)/$$cat as" \
 		   "$$dir/$(PACKAGE)$(INSTOBJEXT).m"; \
 	    else \
--- ./gnokii/Makefile.orig	Sun Jul  1 14:08:44 2001
+++ ./gnokii/Makefile	Sun Jul  1 14:09:34 2001
@@ -40,8 +40,8 @@
 	$(RM) $(OBJS) *~ depend gnokii *.exe core *.bak
 
 install:
-	$(INSTALL) -d $(bindir)
-	$(INSTALL) gnokii $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) gnokii $(DESTDIR)$(bindir)
 
 depend dep:
 	$(CC) $(CFLAGS) -MM *.c >depend
--- ./gnokiid/Makefile.orig	Sun Jul  1 14:09:56 2001
+++ ./gnokiid/Makefile	Sun Jul  1 14:10:30 2001
@@ -28,8 +28,8 @@
 	$(RM) $(OBJS) *~ depend gnokiid *.exe core *.bak
 
 install:
-	$(INSTALL) -d $(sbindir)
-	$(INSTALL) gnokiid $(sbindir)
+	$(INSTALL) -d $(DESTDIR)$(sbindir)
+	$(INSTALL) gnokiid $(DESTDIR)$(sbindir)
 
 depend dep:
 	$(CC) $(CFLAGS) -MM *.c >depend
--- ./utils/Makefile.orig	Sun Jul  1 14:13:45 2001
+++ ./utils/Makefile	Sun Jul  1 14:28:09 2001
@@ -19,9 +19,9 @@
 	$(RM) $(OBJS) *~ depend mgnokiidev *.exe core *.bak
 
 install:
-	$(INSTALL) -d $(sbindir)
-	$(INSTALL) -o root -g gnokii -m 4750 mgnokiidev $(sbindir)
-	$(INSTALL) todologo $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(sbindir)
+	$(INSTALL) mgnokiidev $(DESTDIR)$(sbindir)
+	$(INSTALL) todologo $(DESTDIR)$(bindir)
 
 depend dep:
 	$(CC) $(CFLAGS) -MM *.c >depend
--- ./xgnokii/Makefile.orig	Sun Jul  1 14:14:46 2001
+++ ./xgnokii/Makefile	Sun Jul  1 14:17:17 2001
@@ -58,27 +58,27 @@
 	$(RM) $(OBJS) *~ xpm/*~ depend xgnokii
 
 install: all
-	$(INSTALL) -d $(bindir)
-	$(INSTALL) -m 755 -s xgnokii $(bindir)
-	$(INSTALL) -d $(xgnokii_libdir)/help
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) -m 755 -s xgnokii $(DESTDIR)$(bindir)
+	$(INSTALL) -d $(DESTDIR)$(xgnokii_libdir)/help
 	( cd $(HELP_DIR); \
 	  $(FIND) . -type d \! -path "*CVS*" \
-	       -exec $(INSTALL) -d $(xgnokii_libdir)/help/{} \; ; \
+	       -exec $(INSTALL) -d $(DESTDIR)$(xgnokii_libdir)/help/{} \; ; \
 	  $(FIND) . -type f \! -path "*CVS*" \
-	       -exec $(INSTALL) -m 0444 {} $(xgnokii_libdir)/help/{} \; \
+	       -exec $(INSTALL) {} $(DESTDIR)$(xgnokii_libdir)/help/{} \; \
 	)
-	$(INSTALL) -m 0444 ../Docs/CREDITS $(xgnokii_libdir)/help/en_US
-	$(INSTALL) -m 0444 ../COPYING $(xgnokii_libdir)/help/en_US
-	$(INSTALL) -d $(xgnokii_libdir)/help/en_US/sample
-	$(INSTALL) -m 0444 ../Docs/sample/gnokiirc $(xgnokii_libdir)/help/en_US/sample
+	$(INSTALL) ../Docs/CREDITS $(DESTDIR)$(xgnokii_libdir)/help/en_US
+	$(INSTALL) ../COPYING $(DESTDIR)$(xgnokii_libdir)/help/en_US
+	$(INSTALL) -d $(DESTDIR)$(xgnokii_libdir)/help/en_US/sample
+	$(INSTALL) ../Docs/sample/gnokiirc $(DESTDIR)$(xgnokii_libdir)/help/en_US/sample
 	(cd $(xgnokii_libdir)/help; ln -sf en_US C; ln -sf en_US POSIX)
-	$(INSTALL) -d $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/6110.xpm $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/6150.xpm $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/5110.xpm $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/Preview_6110.xpm $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/Preview_6150.xpm $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/Preview_6210.xpm $(xgnokii_libdir)/xpm
+	$(INSTALL) -d $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/6110.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/6150.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/5110.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/Preview_6110.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/Preview_6150.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/Preview_6210.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
 
 depend dep:
 	$(CC) $(CFLAGS) -MM *.c >depend
--- ./xlogos/Makefile.orig	Sun Jul  1 14:17:46 2001
+++ ./xlogos/Makefile	Sun Jul  1 15:20:23 2001
@@ -36,8 +36,8 @@
 	$(RM) $(OBJS) *~ depend xlogos *.exe core *.bak
 
 install: all
-	$(INSTALL) -d $(bindir)
-	$(INSTALL) xlogos $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(prefix)/X11R6/bin
+	$(INSTALL) xlogos $(DESTDIR)$(prefix)/X11R6/bin
 
 depend dep:
 	$(CC) $(CFLAGS) -MM *.c >depend
--- ./Makefile.orig	Sun Jul  1 14:04:15 2001
+++ ./Makefile	Sun Jul  1 14:25:19 2001
@@ -8,12 +8,12 @@
 #               2000 Karel Zak
 #
 
-TOPDIR=.
+DESTDIR=
 
 #
 # Makefile.global contains gnokii global settings
 #
-include ${TOPDIR}/Makefile.global
+include ./Makefile.global
 
 
 DIRS =  common/phones \
--- ./Makefile.global.in.orig	Sun Jul  1 14:04:57 2001
+++ ./Makefile.global.in	Sun Jul  1 14:22:04 2001
@@ -25,7 +25,7 @@
 bindir         = @bindir@
 sbindir        = @sbindir@
 libdir         = @libdir@/${PACKAGE}
-xgnokii_libdir = @XGNOKIIDIR@/@XPACKAGE@
+xgnokii_libdir = ${prefix}/X11R6/lib/@XPACKAGE@
 mandir         = @mandir@/man1/
 docdir         = ${prefix}/doc/${PACKAGE}
 locdir         = @datadir@/locale
--- ./configure.in.orig	Sun Jul  1 14:06:45 2001
+++ ./configure.in	Sun Jul  1 14:23:08 2001
@@ -15,7 +15,6 @@
 AC_PREFIX_DEFAULT("/usr/local")
 
 dnl ======================== Default setting
-CFLAGS="-O2 -Wall"
 ALL_LINGUAS="cs de et fi nl sk pl it"
 
 dnl ======================== Checks for programs.
@@ -33,7 +32,7 @@
 
 AC_ARG_ENABLE(debug, 
    [  --enable-debug          compile with debug code],
-   [ CFLAGS="-g -Wall"
+   [ CFLAGS="$CFLAGS -g -Wall"
      AC_DEFINE(DEBUG)
      debug="yes" ],
    [ debug="no"  ]
