diff -ruN gnokii-0.4.0.org/Docs/sample/gnokiirc gnokii-0.4.0/Docs/sample/gnokiirc
--- gnokii-0.4.0.org/Docs/sample/gnokiirc	Wed Jul 10 00:49:13 2002
+++ gnokii-0.4.0/Docs/sample/gnokiirc	Thu Jul 11 21:22:05 2002
@@ -31,7 +31,7 @@
 # In particular ensure that mgnokiidev is in this location, with
 # permissions 4750, owned by root, group gnokii.  Ensure you
 # are in the gnokii group and that the group exists...
-bindir = /usr/local/sbin/
+bindir = /usr/sbin/
 
 # Set this to 'yes' if you want gnokii to set and check the lock file in
 # /var/lock directory. It avoids potential conflicts with other serial
