diff -Nur gnokii-0.5.5.orig/autogen.sh gnokii-0.5.5/autogen.sh
--- gnokii-0.5.5.orig/autogen.sh	2003-10-08 18:31:01.000000000 +0200
+++ gnokii-0.5.5/autogen.sh	2003-10-27 23:33:29.945492448 +0100
@@ -12,6 +12,7 @@
 	mv configure.in~ configure.in
 fi
 
+libtoolize -c -f
 aclocal
 autoheader
 autoconf
diff -Nur gnokii-0.5.5.orig/common/data/Makefile gnokii-0.5.5/common/data/Makefile
--- gnokii-0.5.5.orig/common/data/Makefile	2003-10-08 18:31:01.000000000 +0200
+++ gnokii-0.5.5/common/data/Makefile	2003-10-27 23:09:16.507448616 +0100
@@ -17,19 +17,19 @@
 TOPDIR=../..
 include $(TOPDIR)/Makefile.global
 
-OBJS =	at-emulator.o \
-	datapump.o \
-	rlp-common.o \
-	rlp-crc24.o \
-	virtmodem.o
+OBJS =	at-emulator.lo \
+	datapump.lo \
+	rlp-common.lo \
+	rlp-crc24.lo \
+	virtmodem.lo
 
-all: DATA.o
+all: libDATA.la
 
-DATA.o: $(OBJS)
-	$(LD) DATA.o $(OBJS)
+libDATA.la: $(OBJS)
+	$(LIBTOOL) --mode=link $(CC) -o libDATA.la $(OBJS)
 
 clean:
-	$(RM) $(OBJS) *~ depend DATA.o
+	$(RM) $(OBJS) *~ depend libDATA.la
 
 install:
 	@echo
diff -Nur gnokii-0.5.5.orig/common/devices/Makefile gnokii-0.5.5/common/devices/Makefile
--- gnokii-0.5.5.orig/common/devices/Makefile	2003-10-08 18:31:01.000000000 +0200
+++ gnokii-0.5.5/common/devices/Makefile	2003-10-27 23:10:10.682212800 +0100
@@ -17,24 +17,24 @@
 TOPDIR=../..
 include $(TOPDIR)/Makefile.global
 
-OBJS =	tekram.o \
-	unixirda.o \
-	unixbluetooth.o
+OBJS =	tekram.lo \
+	unixirda.lo \
+	unixbluetooth.lo
 
 ifdef WIN32
-	OBJS += winserial.o
+	OBJS += winserial.lo
 else
-	OBJS += unixserial.o \
-		tcp.o
+	OBJS += unixserial.lo \
+		tcp.lo
 endif
 
-all: DEVICES.o
+all: libDEVICES.la
 
-DEVICES.o: $(OBJS)
-	$(LD) DEVICES.o $(OBJS)
+libDEVICES.la: $(OBJS)
+	$(LIBTOOL) --mode=link $(CC) -o libDEVICES.la $(OBJS)
 
 clean:
-	$(RM) $(OBJS) *~ depend LINKS.o
+	$(RM) $(OBJS) *~ depend libDEVICES.la
 
 install:
 	@echo
diff -Nur gnokii-0.5.5.orig/common/links/Makefile gnokii-0.5.5/common/links/Makefile
--- gnokii-0.5.5.orig/common/links/Makefile	2003-10-08 18:31:02.000000000 +0200
+++ gnokii-0.5.5/common/links/Makefile	2003-10-27 23:12:00.170568040 +0100
@@ -17,22 +17,22 @@
 TOPDIR=../..
 include $(TOPDIR)/Makefile.global
 
-OBJS =	atbus.o \
-	fbus.o \
-	fbus-phonet.o \
-	fbus-3110.o \
-	m2bus.o \
-	utils.o
+OBJS =	atbus.lo \
+	fbus.lo \
+	fbus-phonet.lo \
+	fbus-3110.lo \
+	m2bus.lo \
+	utils.lo
 
-#	cbus.o
+#	cbus.lo
 
-all: LINKS.o
+all: libLINKS.la
 
-LINKS.o: $(OBJS)
-	$(LD) LINKS.o $(OBJS)
+libLINKS.la: $(OBJS)
+	$(LIBTOOL) --mode=link $(CC) -o libLINKS.la $(OBJS)
 
 clean:
-	$(RM) $(OBJS) *~ depend LINKS.o
+	$(RM) $(OBJS) *~ depend libLINKS.la
 
 install:
 	@echo
diff -Nur gnokii-0.5.5.orig/common/Makefile gnokii-0.5.5/common/Makefile
--- gnokii-0.5.5.orig/common/Makefile	2003-10-08 18:31:01.000000000 +0200
+++ gnokii-0.5.5/common/Makefile	2003-10-27 23:20:26.539588248 +0100
@@ -19,92 +19,66 @@
 
 MAJOR_NUMBER = 1
 MINOR_NUMBER = 4
+LIBTOOL_VER = -version-info $(MAJOR_NUMBER):$(MINOR_NUMBER):0
 
-OBJS =	gsm-api.o \
-	gsm-error.o \
-	gsm-statemachine.o \
-	cfgreader.o \
-	device.o \
-	vcal.o \
-	vcard.o \
-	ldif.o \
-	gsm-networks.o \
-	gsm-filetypes.o \
-	readmidi.o \
-	gsm-ringtones.o \
-	gsm-bitmaps.o \
-	gsm-sms.o \
-	gsm-call.o \
-	gsm-encoding.o \
-	gsm-common.o \
-	sms-nokia.o \
-	nokia-decoding.o \
-	compat.o \
-	misc.o \
-	snprintf.o
+OBJS =	gsm-api.lo \
+	gsm-error.lo \
+	gsm-statemachine.lo \
+	cfgreader.lo \
+	device.lo \
+	vcal.lo \
+	vcard.lo \
+	ldif.lo \
+	gsm-networks.lo \
+	gsm-filetypes.lo \
+	readmidi.lo \
+	gsm-ringtones.lo \
+	gsm-bitmaps.lo \
+	gsm-sms.lo \
+	gsm-call.lo \
+	gsm-encoding.lo \
+	gsm-common.lo \
+	sms-nokia.lo \
+	nokia-decoding.lo \
+	compat.lo \
+	misc.lo \
+	snprintf.lo
 
-ifndef WIN32_CROSS
 ifdef NO_SHARED
-GNOKII_LIB=libgnokii.a
+LIBTOOL_RPATH=
 else
-GNOKII_LIB  =libgnokii.so
-GNOKII_LIBM =$(GNOKII_LIB).$(MAJOR_NUMBER)
-GNOKII_LIBMM=$(GNOKII_LIBM).$(MINOR_NUMBER)
+LIBTOOL_RPATH=-rpath $(libdir)
 endif
-else
-ifdef NO_SHARED
-GNOKII_LIB=gnokii.lib
-else
-GNOKII_LIB=gnokii.dll
-endif
-endif
-
-all: $(GNOKII_LIB)
-
-phones/PHONES.o:
-	$(MAKE) -C phones PHONES.o
-
-links/LINKS.o:
-	$(MAKE) -C links LINKS.o
 
-devices/DEVICES.o:
-	$(MAKE) -C devices DEVICES.o
+all: libgnokii.la
 
-gnokii-debug: $(OBJS) phones/PHONES.o links/LINKS.o devices/DEVICES.o ../gnokii/gnokii.o
-	$(CC) -o gnokii-debug $(OBJS) phones/PHONES.o links/LINKS.o devices/DEVICES.o ../gnokii/gnokii.o
+phones/libPHONES.la:
+	$(MAKE) -C phones libPHONES.la
 
-libgnokii.so: $(OBJS) phones/PHONES.o links/LINKS.o devices/DEVICES.o
-	$(CC) -shared -Wl,-soname,$(GNOKII_LIBM) -o $(GNOKII_LIBMM) $(OBJS) phones/PHONES.o links/LINKS.o devices/DEVICES.o
-	ln -sf $(GNOKII_LIBMM) $(GNOKII_LIBM)
-	ln -sf $(GNOKII_LIBM) $(GNOKII_LIB)
+links/libLINKS.la:
+	$(MAKE) -C links libLINKS.la
 
-libgnokii.a: $(OBJS) phones/PHONES.o links/LINKS.o devices/DEVICES.o
-	$(RM) libgnokii.a
-	$(AR) rc libgnokii.a $(OBJS) phones/PHONES.o links/LINKS.o devices/DEVICES.o 
-	$(RANLIB) libgnokii.a
+devices/libDEVICES.la:
+	$(MAKE) -C devices libDEVICES.la
 
-gnokii.dll: $(OBJS) phones/PHONES.o links/LINKS.o devices/DEVICES.o
-	$(CC) -mwindows -shared -o $@ $(OBJS) phones/PHONES.o links/LINKS.o devices/DEVICES.o -Wl,--output-def,gnokii.def,--out-implib,gnokii.lib
+gnokii-debug: $(OBJS) phones/libPHONES.la links/libLINKS.la devices/libDEVICES.la ../gnokii/gnokii.o
+	$(LIBTOOL) --mode=link $(CC) -o gnokii-debug $(OBJS) phones/libPHONES.la links/libLINKS.la devices/libDEVICES.la ../gnokii/gnokii.o
 
-gnokii.lib: $(OBJS) phones/PHONES.o links/LINKS.o devices/DEVICES.o
-	$(RM) gnokii.lib
-	$(AR) rc gnokii.lib $(OBJS) phones/PHONES.o links/LINKS.o devices/DEVICES.o
-	$(RANLIB) gnokii.lib
+libgnokii.la: $(OBJS) phones/libPHONES.la links/libLINKS.la devices/libDEVICES.la
+	$(LIBTOOL) --mode=link $(CC) -o libgnokii.la $(OBJS) phones/libPHONES.la links/libLINKS.la devices/libDEVICES.la $(LIBTOOL_VER) $(LIBTOOL_RPATH)
 
 vcal.c: vcal.lx
 	$(LEX) -ovcal.c vcal.lx
 
 clean:
-	$(RM) $(OBJS) *~ depend libgnokii.a libgnokii.so* gnokii.lib gnokii.dll gnokii.def phones/PHONES.o links/LINKS.o devices/DEVICES.o
+	$(RM) $(OBJS) *~ depend libgnokii.la phones/libPHONES.la links/libLINKS.la devices/libDEVICES.la
 
 distclean:
 	$(RM) vcal.c
 
 install:
 	$(INSTALL) -d $(DESTDIR)$(libdir)
-	$(INSTALL) $(GNOKII_LIBMM) $(DESTDIR)$(libdir)
-	cd $(DESTDIR)$(libdir) && ln -sf $(GNOKII_LIBMM) $(GNOKII_LIBM)
-	cd $(DESTDIR)$(libdir) && ln -sf $(GNOKII_LIBM) $(GNOKII_LIB)
+	$(LIBTOOL) --mode=install $(INSTALL) libgnokii.la $(DESTDIR)$(libdir)
 	$(INSTALL) -d $(DESTDIR)$(libdir)/pkgconfig
 	$(INSTALL_DATA) gnokii.pc $(DESTDIR)$(libdir)/pkgconfig
 	@echo
diff -Nur gnokii-0.5.5.orig/common/phones/Makefile gnokii-0.5.5/common/phones/Makefile
--- gnokii-0.5.5.orig/common/phones/Makefile	2003-10-08 18:31:02.000000000 +0200
+++ gnokii-0.5.5/common/phones/Makefile	2003-10-27 23:11:52.538728256 +0100
@@ -17,30 +17,30 @@
 TOPDIR=../..
 include $(TOPDIR)/Makefile.global
 
-OBJS =	generic.o \
-	nokia.o \
-	nk7110.o \
-	nk6510.o \
-	nk6160.o \
-	nk6100.o \
-	nk3110.o \
-	atgen.o \
-	atbosch.o \
-	ateric.o \
-	atnok.o \
-	atsie.o \
-	fake.o
+OBJS =	generic.lo \
+	nokia.lo \
+	nk7110.lo \
+	nk6510.lo \
+	nk6160.lo \
+	nk6100.lo \
+	nk3110.lo \
+	atgen.lo \
+	atbosch.lo \
+	ateric.lo \
+	atnok.lo \
+	atsie.lo \
+	fake.lo
 
-#	nk2110.o
-#	dc2711.o
+#	nk2110.lo
+#	dc2711.lo
 
-all: PHONES.o
+all: libPHONES.la
 
-PHONES.o: $(OBJS)
-	$(LD) PHONES.o $(OBJS)
+libPHONES.la: $(OBJS)
+	$(LIBTOOL) --mode=link $(CC) -o libPHONES.la $(OBJS)
 
 clean:
-	$(RM) $(OBJS) *~ depend PHONES.o
+	$(RM) $(OBJS) *~ depend libPHONES.la
 
 install:
 	@echo
diff -Nur gnokii-0.5.5.orig/configure.in gnokii-0.5.5/configure.in
--- gnokii-0.5.5.orig/configure.in	2003-10-08 18:31:01.000000000 +0200
+++ gnokii-0.5.5/configure.in	2003-10-27 22:52:05.894125656 +0100
@@ -23,7 +23,7 @@
 AC_PATH_PROG(FIND, find, no)
 AC_CHECK_PROGS(MAKE, gmake make)
 AC_CHECK_PROGS(AR, ar)
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 AC_PROG_INSTALL
 
 AM_ICONV
@@ -319,16 +319,12 @@
    [ security="no"  ]
 )
 
-dnl TODO: We should be checking for GNU ld and only using GNU specific
-dnl paramaters if it's found.
-AC_ARG_ENABLE(shared,
-   [  --disable-shared        build without shared library ],
-   [ NO_SHARED=1
-     shared="no"
-     GLDFLAGS="" ],
-   [ shared="yes"
-     GLDFLAGS=""]
-)
+dnl LIBTOOL check handles --enable-shared/--disable-shared
+if test x$enable_shared = xyes ; then
+	shared="yes"
+else
+	shared="no"
+fi
 
 AC_ARG_ENABLE(win32, 
    [  --enable-win32          if you want win32 suport ],
diff -Nur gnokii-0.5.5.orig/gnokii/Makefile gnokii-0.5.5/gnokii/Makefile
--- gnokii-0.5.5.orig/gnokii/Makefile	2003-10-08 18:31:02.000000000 +0200
+++ gnokii-0.5.5/gnokii/Makefile	2003-10-27 23:23:01.017104096 +0100
@@ -16,22 +16,12 @@
 endif
 
 ifndef WIN32_CROSS
-ifdef NO_SHARED
-	GNOKII_LIB=libgnokii.a
-else
-	GNOKII_LIB=libgnokii.so
-endif
 PRGNAME=gnokii
 else
-ifdef NO_SHARED
-	GNOKII_LIB=gnokii.lib
-else
-	GNOKII_LIB=gnokii.dll
-endif
 PRGNAME=gnokii.exe
 endif
 
-LDLIBS += -L$(TOPDIR)/common -lgnokii
+LDLIBS += $(TOPDIR)/common/libgnokii.la
 
 ifdef XPM_LIBS
 	LDLIBS += $(XPM_LIBS)
@@ -39,11 +29,11 @@
 
 all: $(PRGNAME)
 
-gnokii: $(OBJS) $(TOPDIR)/common/$(GNOKII_LIB)
-	$(CC) $(LDFLAGS) $(OBJS) $(LDLIBS) -o $@
+gnokii: $(OBJS) $(TOPDIR)/common/libgnokii.la
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(OBJS) $(LDLIBS) -o $@
 
 gnokii.exe: $(OBJS) $(TOPDIR)/common/$(GNOKII_LIB)
-	$(CC) -mconsole $(LDFLAGS) $(OBJS) $(LDLIBS) -o $@
+	$(LIBTOOL) --mode=link $(CC) -mconsole $(LDFLAGS) $(OBJS) $(LDLIBS) -o $@
 
 $(TOPDIR)/getopt/libgetopt.a:
 	$(MAKE) -C $(TOPDIR)/getopt
@@ -53,11 +43,11 @@
 
 install: all
 	$(INSTALL) -d $(DESTDIR)$(bindir)
-	$(INSTALL) gnokii $(DESTDIR)$(bindir)
+	$(LIBTOOL) --mode=install $(INSTALL) gnokii $(DESTDIR)$(bindir)
 
 install-strip: all
 	$(INSTALL) -d $(DESTDIR)$(bindir)
-	$(INSTALL) -s gnokii $(DESTDIR)$(bindir)
+	$(LIBTOOL) --mode=install $(INSTALL) -s gnokii $(DESTDIR)$(bindir)
 	@echo "done"
 
 install-suid: all
@@ -65,7 +55,7 @@
 		/usr/sbin/groupadd gnokii; \
 	fi
 	$(INSTALL) -d $(DESTDIR)$(bindir)
-	$(INSTALL) -o root -g gnokii -m 0750 gnokii $(DESTDIR)$(bindir)
+	$(LIBTOOL) --mode=install $(INSTALL) -o root -g gnokii -m 0750 gnokii $(DESTDIR)$(bindir)
 	@echo
 
 install-ss: all
@@ -73,7 +63,7 @@
 		/usr/sbin/groupadd gnokii; \
 	fi
 	$(INSTALL) -d $(DESTDIR)$(bindir)
-	$(INSTALL) -o root -g gnokii -m 0750 -s gnokii $(DESTDIR)$(bindir)
+	$(LIBTOOL) --mode=install $(INSTALL) -o root -g gnokii -m 0750 -s gnokii $(DESTDIR)$(bindir)
 	@echo
 
 depend dep:
diff -Nur gnokii-0.5.5.orig/gnokiid/Makefile gnokii-0.5.5/gnokiid/Makefile
--- gnokii-0.5.5.orig/gnokiid/Makefile	2003-10-08 18:31:02.000000000 +0200
+++ gnokii-0.5.5/gnokiid/Makefile	2003-10-27 23:24:02.385774640 +0100
@@ -9,7 +9,7 @@
 TOPDIR=..
 include $(TOPDIR)/Makefile.global
 
-LDLIBS += -L$(TOPDIR)/common -lgnokii
+LDLIBS += $(TOPDIR)/common/libgnokii.la
 
 ifdef XPM_LIBS
 	LDLIBS += $(XPM_LIBS)
@@ -17,27 +17,21 @@
 
 OBJS = gnokiid.o
 
-ifdef NO_SHARED
-        GNOKII_LIB=libgnokii.a
-else
-        GNOKII_LIB=libgnokii.so
-endif
-
 all: gnokiid
 
-gnokiid: $(OBJS) $(TOPDIR)/common/data/DATA.o $(TOPDIR)/common/$(GNOKII_LIB)
-	$(CC) $(LDFLAGS) $(OBJS) $(TOPDIR)/common/data/DATA.o $(LDLIBS) -o $@
+gnokiid: $(OBJS) $(TOPDIR)/common/data/libDATA.la $(TOPDIR)/common/libgnokii.la
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(OBJS) $(TOPDIR)/common/data/libDATA.la $(LDLIBS) -o $@
 
 clean:
 	$(RM) $(OBJS) *~ depend gnokiid *.exe core *.bak
 
 install: all
 	$(INSTALL) -d $(DESTDIR)/$(sbindir)
-	$(INSTALL) gnokiid $(DESTDIR)/$(sbindir)
+	$(LIBTOOL) --mode=install $(INSTALL) gnokiid $(DESTDIR)/$(sbindir)
 
 install-strip: all
 	$(INSTALL) -d $(DESTDIR)/$(sbindir)
-	$(INSTALL) -s gnokiid $(DESTDIR)/$(sbindir)
+	$(LIBTOOL) --mode=install $(INSTALL) -s gnokiid $(DESTDIR)/$(sbindir)
 	@echo "done"
 
 install-suid: all
@@ -45,7 +39,7 @@
 		/usr/sbin/groupadd gnokii; \
 	fi
 	$(INSTALL) -d $(DESTDIR)/$(sbindir)
-	$(INSTALL) -o root -g gnokii -m 4750 gnokiid $(DESTDIR)/$(sbindir)
+	$(LIBTOOL) --mode=install $(INSTALL) -o root -g gnokii -m 4750 gnokiid $(DESTDIR)/$(sbindir)
 	@echo "done"
 
 install-ss: all
@@ -53,7 +47,7 @@
 		/usr/sbin/groupadd gnokii; \
 	fi
 	$(INSTALL) -d $(DESTDIR)/$(sbindir)
-	$(INSTALL) -o root -g gnokii -m 4750 -s gnokiid $(DESTDIR)/$(sbindir)
+	$(LIBTOOL) --mode=install $(INSTALL) -o root -g gnokii -m 4750 -s gnokiid $(DESTDIR)/$(sbindir)
 	@echo "done"
 
 depend dep:
diff -Nur gnokii-0.5.5.orig/Makefile.global.in gnokii-0.5.5/Makefile.global.in
--- gnokii-0.5.5.orig/Makefile.global.in	2003-10-08 18:31:02.000000000 +0200
+++ gnokii-0.5.5/Makefile.global.in	2003-10-27 23:14:13.191345808 +0100
@@ -12,8 +12,6 @@
 
 SHELL          = @SHELL@
 
-LD             = @CC@ -Xlinker -r -nostdlib -nodefaultlibs -o
-
 PACKAGE        = gnokii
 
 srcdir         = @srcdir@
@@ -51,6 +49,8 @@
 LEX            = @LEX@
 AR             = @AR@
 RANLIB         = @RANLIB@
+top_builddir   = $(TOPDIR)
+LIBTOOL        = @LIBTOOL@
 
 GTK_CFLAGS     = @GTK_CFLAGS@
 GTK_LIBS       = @GTK_LIBS@
@@ -87,3 +87,6 @@
 
 CFLAGS += -I$(GNOKII_INCLUDE)
 LDLIBS = $(LIBS)
+
+%.lo: %.c
+	$(LIBTOOL) --mode=compile $(CC) $(CFLAGS) $(CPPFLAGS) -o $@ -c $<
diff -Nur gnokii-0.5.5.orig/smsd/Makefile gnokii-0.5.5/smsd/Makefile
--- gnokii-0.5.5.orig/smsd/Makefile	2003-10-08 18:31:01.000000000 +0200
+++ gnokii-0.5.5/smsd/Makefile	2003-10-27 23:29:28.112256688 +0100
@@ -26,7 +26,7 @@
 LDLIBS += $(PTHREAD_LIBS) \
 	  $(shell $(GLIBLDLIBS))
 
-LDLIBS += -s -rdynamic -L$(TOPDIR)/common -lgnokii -ldl
+LDLIBS += -rdynamic $(TOPDIR)/common/libgnokii.la -ldl
 # Some systems, eg. FreeBSD don't have libdl. Use this:
 # LDLIBS += -s -rdynamic -L$(TOPDIR)/common -lgnokii
 
@@ -40,8 +40,8 @@
 all: smsd
 
 
-smsd: $(OBJS) $(TOPDIR)/common/libgnokii.so
-	gcc $(LDFLAGS) $(OBJS) $(LDLIBS) -o $@
+smsd: $(OBJS) $(TOPDIR)/common/libgnokii.la
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(OBJS) $(LDLIBS) -o $@
 
 
 ### DB Modules
@@ -49,33 +49,35 @@
 # FIXME: detect it in configure
 
 # PostgreSQL support
-libpq.so: smsd.h pq.c
-	gcc -s -shared -o libpq.so $(CFLAGS) -I../include \
-	-I/usr/include/pgsql pq.c $(shell $(GLIBLDLIBS)) -lpq
+libpq.la: smsd.h pq.c
+	$(LIBTOOL) --mode=link $(CC) -o libpq.la $(CFLAGS) -I../include \
+	-I/usr/include/pgsql pq.c $(shell $(GLIBLDLIBS)) -lpq -avoid-version -rpath $(libdir)/smsd
 
 # MySQL support
-libmysql.so: smsd.h mysql.c
-	gcc -s -shared -o libmysql.so $(CFLAGS) -I../include \
+libmysql.la: smsd.h mysql.c
+	$(LIBTOOL) --mode=link $(CC) -o libmysql.la $(CFLAGS) -I../include \
 	$(shell mysql_config --cflags) mysql.c $(shell $(GLIBLDLIBS)) $(shell mysql_config --libs)
 
 # File support
-libfile.so: smsd.h file.c
-	gcc -s -shared -o libfile.so $(CFLAGS) -I../include \
+libfile.la: smsd.h file.c
+	$(LIBTOOL) --mode=link $(CC) -o libfile.so $(CFLAGS) -I../include \
         file.c $(shell $(GLIBLDLIBS))
 
 ### End of DB Modules
 
-$(TOPDIR)/common/libgnokii.so:
-	$(MAKE) -C $(TOPDIR)/common libgnokii.so
+$(TOPDIR)/common/libgnokii.la:
+	$(MAKE) -C $(TOPDIR)/common libgnokii.la
 
 clean:
 	$(RM) $(OBJS) *~ depend smsd *.so core*
 
 install: all
 	$(INSTALL) -d $(DESTDIR)$(sbindir)
-	$(INSTALL) -m 755 -s smsd $(DESTDIR)$(sbindir)
+	$(LIBTOOL) --mode=install $(INSTALL) -m 755 smsd $(DESTDIR)$(sbindir)
 	$(INSTALL) -d $(DESTDIR)${libdir}/smsd
-	$(INSTALL) lib*.so $(DESTDIR)${libdir}/smsd
+	for f in lib*.la ; do \
+	$(LIBTOOL) --mode=install $(INSTALL) $$f $(DESTDIR)${libdir}/smsd ; \
+	done
 	$(INSTALL) -d $(DESTDIR)$(man8dir)
 	$(INSTALL_DATA) $(SMSD_MAN) $(DESTDIR)$(man8dir)
 
diff -Nur gnokii-0.5.5.orig/xgnokii/Makefile gnokii-0.5.5/xgnokii/Makefile
--- gnokii-0.5.5.orig/xgnokii/Makefile	2003-10-08 18:31:03.000000000 +0200
+++ gnokii-0.5.5/xgnokii/Makefile	2003-10-27 23:32:06.807131400 +0100
@@ -23,43 +23,37 @@
 LDLIBS += $(PTHREAD_LIBS) \
 	  $(GTK_LIBS)
 
-GLDLIBS += -L$(TOPDIR)/common -lgnokii
+GLDLIBS += $(TOPDIR)/common/libgnokii.la
 
-OBJS = 	xgnokii.o \
-	xgnokii_common.o \
-	xgnokii_contacts.o \
-	xgnokii_sms.o \
-	xgnokii_netmon.o \
-	xgnokii_dtmf.o \
-	xgnokii_cfg.o \
-	xgnokii_speed.o \
-	xgnokii_xkeyb.o \
-	xgnokii_calendar.o \
-	xgnokii_logos.o \
-	xgnokii_xring.o \
-	xgnokii_lowlevel.o \
-	xgnokii_data.o
+OBJS = 	xgnokii.lo \
+	xgnokii_common.lo \
+	xgnokii_contacts.lo \
+	xgnokii_sms.lo \
+	xgnokii_netmon.lo \
+	xgnokii_dtmf.lo \
+	xgnokii_cfg.lo \
+	xgnokii_speed.lo \
+	xgnokii_xkeyb.lo \
+	xgnokii_calendar.lo \
+	xgnokii_logos.lo \
+	xgnokii_xring.lo \
+	xgnokii_lowlevel.lo \
+	xgnokii_data.lo
 
 HELP_DIR = docs/help
 
-ifdef NO_SHARED
-        GNOKII_LIB=libgnokii.a
-else
-        GNOKII_LIB=libgnokii.so
-endif
-
 all: xgnokii
 
 
-xgnokii: $(OBJS) $(TOPDIR)/common/data/DATA.o $(TOPDIR)/common/$(GNOKII_LIB)
-	$(CC) $(LDFLAGS) $(OBJS) $(GLDLIBS) $(TOPDIR)/common/data/DATA.o $(LDLIBS) -o $@
+xgnokii: $(OBJS) $(TOPDIR)/common/data/libDATA.la $(TOPDIR)/common/libgnokii.la
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(OBJS) $(GLDLIBS) $(TOPDIR)/common/data/libDATA.la $(LDLIBS) -o $@
 
 clean:
 	$(RM) $(OBJS) *~ xpm/*~ depend xgnokii
 
 install: all
 	$(INSTALL) -d $(DESTDIR)$(xbindir)
-	$(INSTALL) xgnokii $(DESTDIR)$(xbindir)
+	$(LIBTOOL) --mode=install $(INSTALL) xgnokii $(DESTDIR)$(xbindir)
 	$(INSTALL) -d $(DESTDIR)$(xgnokii_libdir)/help
 	( cd $(HELP_DIR); \
 	  $(FIND) . -type d \! -path "*CVS*" \
@@ -89,7 +83,7 @@
 
 install-strip: all
 	$(INSTALL) -d $(DESTDIR)$(xbindir)
-	$(INSTALL) -s xgnokii $(DESTDIR)$(xbindir)
+	$(LIBTOOL) --mode=install $(INSTALL) -s xgnokii $(DESTDIR)$(xbindir)
 	$(INSTALL) -d $(DESTDIR)$(xgnokii_libdir)/help
 	( cd $(HELP_DIR); \
 	  $(FIND) . -type d \! -path "*CVS*" \
@@ -123,7 +117,7 @@
 		/usr/sbin/groupadd gnokii; \
 	fi
 	$(INSTALL) -d $(DESTDIR)$(xbindir)
-	$(INSTALL) -o root -g gnokii -m 0750 xgnokii $(DESTDIR)$(xbindir)
+	$(LIBTOOL) --mode=install $(INSTALL) -o root -g gnokii -m 0750 xgnokii $(DESTDIR)$(xbindir)
 	$(INSTALL) -o root -g gnokii -d -m 0750 $(DESTDIR)$(xgnokii_libdir)/help
 	( cd $(HELP_DIR); \
 	  $(FIND) . -type d \! -path "*CVS*" \
@@ -157,7 +151,7 @@
 		/usr/sbin/groupadd gnokii; \
 	fi
 	$(INSTALL) -o root -g gnokii -d $(DESTDIR)$(xbindir)
-	$(INSTALL) -o root -g gnokii -m 0750 -s xgnokii $(DESTDIR)$(xbindir)
+	$(LIBTOOL) --mode=install $(INSTALL) -o root -g gnokii -m 0750 -s xgnokii $(DESTDIR)$(xbindir)
 	$(INSTALL) -o root -g gnokii -d -m 0750 $(DESTDIR)$(xgnokii_libdir)/help
 	( cd $(HELP_DIR); \
 	  $(FIND) . -type d \! -path "*CVS*" \
