diff -Naur gnokii-0.3.3_pre7/Docs/sample/sample.gnokiirc gnokii-0.3.3_pre7-p/Docs/sample/sample.gnokiirc
--- gnokii-0.3.3_pre7/Docs/sample/sample.gnokiirc	Mon Oct 16 18:15:36 2000
+++ gnokii-0.3.3_pre7-p/Docs/sample/sample.gnokiirc	Tue Oct 17 14:44:16 2000
@@ -28,4 +28,4 @@
 # In particular ensure that mgnokiidev is in this location, with
 # permissions 4754, owned by root, group gnokii.  Ensure you
 # are in the gnokii group and that the group exists...
-bindir = /usr/local/sbin/
+bindir = /usr/sbin/
diff -Naur gnokii-0.3.3_pre7/Docs/sample.gnokiirc gnokii-0.3.3_pre7-p/Docs/sample.gnokiirc
--- gnokii-0.3.3_pre7/Docs/sample.gnokiirc	Mon Oct 16 18:15:36 2000
+++ gnokii-0.3.3_pre7-p/Docs/sample.gnokiirc	Tue Oct 17 14:44:16 2000
@@ -28,4 +28,4 @@
 # In particular ensure that mgnokiidev is in this location, with
 # permissions 4754, owned by root, group gnokii.  Ensure you
 # are in the gnokii group and that the group exists...
-bindir = /usr/local/sbin/
+bindir = /usr/sbin/
diff -Naur gnokii-0.3.3_pre7/Makefile gnokii-0.3.3_pre7-p/Makefile
--- gnokii-0.3.3_pre7/Makefile	Mon Oct 16 18:15:36 2000
+++ gnokii-0.3.3_pre7-p/Makefile	Tue Oct 17 14:44:16 2000
@@ -7,6 +7,7 @@
 #
 
 TOPDIR=.
+DESTDIR=
 
 #
 # Makefile.global contains gnokii global settings
diff -Naur gnokii-0.3.3_pre7/Makefile.global.in gnokii-0.3.3_pre7-p/Makefile.global.in
--- gnokii-0.3.3_pre7/Makefile.global.in	Mon Oct 16 18:15:36 2000
+++ gnokii-0.3.3_pre7-p/Makefile.global.in	Tue Oct 17 14:44:16 2000
@@ -31,7 +31,7 @@
 bindir		= @bindir@
 sbindir		= @sbindir@
 libdir		= @libdir@/${PACKAGE}
-xgnokii_libdir	= @libdir@/${XPACKAGE}
+xgnokii_libdir	= ${prefix}/X11R6/lib/${XPACKAGE}
 mandir		= @mandir@/man1/
 docdir		= ${prefix}/doc/${PACKAGE}
 locdir		= @datadir@/locale
diff -Naur gnokii-0.3.3_pre7/configure.in gnokii-0.3.3_pre7-p/configure.in
--- gnokii-0.3.3_pre7/configure.in	Mon Oct 16 18:15:36 2000
+++ gnokii-0.3.3_pre7-p/configure.in	Tue Oct 17 14:45:41 2000
@@ -12,7 +12,6 @@
 AC_CANONICAL_SYSTEM
 
 dnl ======================== Default setting
-CFLAGS="-O2 -Wall"
 ALL_LINGUAS="cs de et fi nl sk pl"
 model="6110"
 case "$build_os" in
@@ -35,7 +34,7 @@
 
 AC_ARG_ENABLE(debug, 
 	[  --enable-debug          compile with debug code],
-	[ CFLAGS="-g -Wall"
+	[ CFLAGS="$CFLAGS -g -Wall"
 	  AC_DEFINE(DEBUG)
           debug="yes" ],
         [ debug="no"  ]
@@ -45,6 +44,7 @@
 	[  --enable-xdebug         compile with xdebug code],
 	[ AC_DEFINE(XDEBUG)
           xdebug="yes" ],
+	a
         [ xdebug="no"  ]
 )
 
diff -Naur gnokii-0.3.3_pre7/gnokii/Makefile gnokii-0.3.3_pre7-p/gnokii/Makefile
--- gnokii-0.3.3_pre7/gnokii/Makefile	Mon Oct 16 18:15:36 2000
+++ gnokii-0.3.3_pre7-p/gnokii/Makefile	Tue Oct 17 14:44:16 2000
@@ -40,8 +40,8 @@
 	$(RM) $(OBJS) *~ depend gnokii *.exe core *.bak
 
 install:
-	$(INSTALL) -d $(bindir)
-	$(INSTALL) gnokii $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) gnokii $(DESTDIR)$(bindir)
 
 depend dep:
 	$(CC) $(CFLAGS) -MM *.c >depend
diff -Naur gnokii-0.3.3_pre7/gnokiid/Makefile gnokii-0.3.3_pre7-p/gnokiid/Makefile
--- gnokii-0.3.3_pre7/gnokiid/Makefile	Mon Oct 16 18:15:36 2000
+++ gnokii-0.3.3_pre7-p/gnokiid/Makefile	Tue Oct 17 14:44:16 2000
@@ -28,8 +28,8 @@
 	$(RM) $(OBJS) *~ depend gnokiid *.exe core *.bak
 
 install:
-	$(INSTALL) -d $(sbindir)
-	$(INSTALL) gnokiid $(sbindir)
+	$(INSTALL) -d $(DESTDIR)$(sbindir)
+	$(INSTALL) gnokiid $(DESTDIR)$(sbindir)
 
 depend dep:
 	$(CC) $(CFLAGS) -MM *.c >depend
diff -Naur gnokii-0.3.3_pre7/po/Makefile.in.in gnokii-0.3.3_pre7-p/po/Makefile.in.in
--- gnokii-0.3.3_pre7/po/Makefile.in.in	Mon Oct 16 18:15:36 2000
+++ gnokii-0.3.3_pre7-p/po/Makefile.in.in	Tue Oct 17 14:44:16 2000
@@ -107,7 +107,7 @@
 install-data: install-data-@USE_NLS@
 install-data-no: all
 install-data-yes: all
-	${SHELL} $(top_srcdir)/mkinstalldirs $(datadir); \
+	${SHELL} $(top_srcdir)/mkinstalldirs $(DESTDIR)$(datadir); \
 	catalogs='$(CATALOGS)'; \
 	for cat in $$catalogs; do \
 	  case "$$cat" in \
@@ -116,22 +116,22 @@
 	  esac; \
 	  lang=`echo $$cat | sed 's/$(CATOBJEXT)$$//'`; \
 	  dir=$$destdir/$$lang/LC_MESSAGES; \
-	  ${SHELL} $(top_srcdir)/mkinstalldirs $$dir; \
+	  ${SHELL} $(top_srcdir)/mkinstalldirs $(DESTDIR)$$dir; \
 	  if test -r $$cat; then \
-	    $(INSTALL_DATA) $$cat $$dir/$(PACKAGE)$(INSTOBJEXT); \
+	    $(INSTALL_DATA) $$cat $(DESTDIR)$$dir/$(PACKAGE)$(INSTOBJEXT); \
 	    echo "installing $$cat as $$dir/$(PACKAGE)$(INSTOBJEXT)"; \
 	  else \
-	    $(INSTALL_DATA) $(srcdir)/$$cat $$dir/$(PACKAGE)$(INSTOBJEXT); \
+	    $(INSTALL_DATA) $(srcdir)/$$cat $(DESTDIR)$$dir/$(PACKAGE)$(INSTOBJEXT); \
 	    echo "installing $(srcdir)/$$cat as" \
 		 "$$dir/$(PACKAGE)$(INSTOBJEXT)"; \
 	  fi; \
 	  if test -r $$cat.m; then \
-	    $(INSTALL_DATA) $$cat.m $$dir/$(PACKAGE)$(INSTOBJEXT).m; \
+	    $(INSTALL_DATA) $$cat.m $(DESTDIR)$$dir/$(PACKAGE)$(INSTOBJEXT).m; \
 	    echo "installing $$cat.m as $$dir/$(PACKAGE)$(INSTOBJEXT).m"; \
 	  else \
 	    if test -r $(srcdir)/$$cat.m ; then \
 	      $(INSTALL_DATA) $(srcdir)/$$cat.m \
-		$$dir/$(PACKAGE)$(INSTOBJEXT).m; \
+		$(DESTDIR)$$dir/$(PACKAGE)$(INSTOBJEXT).m; \
 	      echo "installing $(srcdir)/$$cat as" \
 		   "$$dir/$(PACKAGE)$(INSTOBJEXT).m"; \
 	    else \
diff -Naur gnokii-0.3.3_pre7/utils/Makefile gnokii-0.3.3_pre7-p/utils/Makefile
--- gnokii-0.3.3_pre7/utils/Makefile	Mon Oct 16 18:15:37 2000
+++ gnokii-0.3.3_pre7-p/utils/Makefile	Tue Oct 17 14:44:16 2000
@@ -19,8 +19,8 @@
 	$(RM) $(OBJS) *~ depend mgnokiidev *.exe core *.bak
 
 install:
-	$(INSTALL) -d $(sbindir)
-	$(INSTALL) mgnokiidev $(sbindir)
+	$(INSTALL) -d $(DESTDIR)$(sbindir)
+	$(INSTALL) mgnokiidev $(DESTDIR)$(sbindir)
 
 depend dep:
 	$(CC) $(CFLAGS) -MM *.c >depend
diff -Naur gnokii-0.3.3_pre7/xgnokii/Makefile gnokii-0.3.3_pre7-p/xgnokii/Makefile
--- gnokii-0.3.3_pre7/xgnokii/Makefile	Mon Oct 16 18:15:38 2000
+++ gnokii-0.3.3_pre7-p/xgnokii/Makefile	Tue Oct 17 14:52:53 2000
@@ -50,26 +50,26 @@
 	$(RM) $(OBJS) *~ xpm/*~ depend xgnokii
 
 install: all
-	$(INSTALL) -d $(bindir)
-	$(INSTALL) -m 755 -s xgnokii $(bindir)
-	$(INSTALL) -d $(xgnokii_libdir)/help
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) xgnokii $(DESTDIR)$(bindir)
+	$(INSTALL) -d $(DESTDIR)$(xgnokii_libdir)/help
 	( cd $(HELP_DIR); \
 	  $(FIND) . -type d \! -path "*CVS*" \
-	       -exec $(INSTALL) -d $(xgnokii_libdir)/help/{} \; ; \
+	       -exec $(INSTALL) -d $(DESTDIR)$(xgnokii_libdir)/help/{} \; ; \
 	  $(FIND) . -type f \! -path "*CVS*" \
-	       -exec $(INSTALL) -m 0444 {} $(xgnokii_libdir)/help/{} \; \
+	       -exec $(INSTALL) $(DESTDIR)$(xgnokii_libdir)/help/{} \; \
 	)
-	$(INSTALL) -m 0444 ../Docs/CREDITS $(xgnokii_libdir)/help/en_US
-	$(INSTALL) -m 0444 ../COPYING $(xgnokii_libdir)/help/en_US
-	$(INSTALL) -d $(xgnokii_libdir)/help/en_US/sample
-	$(INSTALL) -m 0444 ../Docs/sample.gnokiirc $(xgnokii_libdir)/help/en_US/sample
+	$(INSTALL) ../Docs/CREDITS $(DESTDIR)$(xgnokii_libdir)/help/en_US
+	$(INSTALL) ../COPYING $(DESTDIR)$(xgnokii_libdir)/help/en_US
+	$(INSTALL) -d $(DESTDIR)$(xgnokii_libdir)/help/en_US/sample
+	$(INSTALL) ../Docs/sample.gnokiirc $(DESTDIR)$(xgnokii_libdir)/help/en_US/sample
 	(cd $(xgnokii_libdir)/help; ln -sf en_US C; ln -sf en_US POSIX)
-	$(INSTALL) -d $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/6110.xpm $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/6150.xpm $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/5110.xpm $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/Preview_6110.xpm $(xgnokii_libdir)/xpm
-	$(INSTALL) -m 0444 xpm/Preview_6150.xpm $(xgnokii_libdir)/xpm
+	$(INSTALL) -d $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/6110.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/6150.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/5110.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/Preview_6110.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
+	$(INSTALL) xpm/Preview_6150.xpm $(DESTDIR)$(xgnokii_libdir)/xpm
 
 depend dep:
 	$(CC) $(CFLAGS) -MM *.c >depend
diff -Naur gnokii-0.3.3_pre7/xlogos/Makefile gnokii-0.3.3_pre7-p/xlogos/Makefile
--- gnokii-0.3.3_pre7/xlogos/Makefile	Mon Oct 16 18:15:39 2000
+++ gnokii-0.3.3_pre7-p/xlogos/Makefile	Tue Oct 17 14:44:16 2000
@@ -36,8 +36,8 @@
 	$(RM) $(OBJS) *~ depend xlogos *.exe core *.bak
 
 install: all
-	$(INSTALL) -d $(bindir)
-	$(INSTALL) xlogos $(bindir) 
+	$(INSTALL) -d $(DESTDIR)$(prefix)/X11R6/bin
+	$(INSTALL) xlogos $(DESTDIR)$(prefix)/X11R6/bin
 
 depend dep:
 	$(CC) $(CFLAGS) -MM *.c >depend
